src/org/sosy_lab/java_smt/solvers/cvc5/CVC5AbstractProver.java

```diff
@@ -103,17 +113,20 @@ protected void popImpl() {
             "You tried to use pop() on an CVC5 assertion stack illegally.", e);
       }
     }
+    assertedTerms.pop();
   }
 
-  @Override
-  protected @Nullable T addConstraintImpl(BooleanFormula pF) throws InterruptedException {
+  @CanIgnoreReturnValue
+  protected String addConstraint0(BooleanFormula pF) {
     Preconditions.checkState(!closed);
     setChanged();
     Term exp = creator.extractInfo(pF);
     if (incremental) {
       solver.assertFormula(exp);
     }
-    return null;
+    String id = "ID_" + ID_GENERATOR.getFreshId();
```

Comment:
I totally get what you are doing here. But we should create a new class (InterpolationPoint?) that wraps this ID to make the intention clear for user and future developers.

Reply:
InterpolationPoint is a quite special use-case. Some solvers also use those references for unsat-cores or model generation. We can think about it independently of this issue. Maybe name the reference to something like FormulaBacktrackPoint.